plugins/woocommerce/includes/admin/helper/class-wc-helper.php

```diff
@@ -1270,11 +1270,24 @@ public static function get_subscriptions() {
 			return $data;
 		}
 
+		$source = $_SERVER['REQUEST_URI'];
+		if ( preg_match( '/\S+wc-addons?\S+/', $source ) ):
+			$source = 'my-subscriptions';
+		elseif ( preg_match( '/\S+plugins.php?\S+/', $source ) ):
+			$source = 'plugins';
+		elseif ( preg_match( '/\S+wc-admin?\S+/', $source ) ):
+			$source = 'inbox-notes';
+		elseif ( preg_match( '/\Sadmin-ajax.php/', $source ) ):
+			$source = 'heartbeat-api';
+		endif;
+
 		// Obtain the connected user info.
 		$request = WC_Helper_API::get(
 			'subscriptions',
 			array(
 				'authenticated' => true,
+				'source'        => '?source=' . $source,
```

Comment:
I reckon we should escape this, just to be safe. It might make sense to name the arg something more generic here, in `WC_Helper_API::request` and `WC_Helper_API::url`, as it's possible someone will want to add other URL params at some point. For example:

```suggestion
				'query_string'        => esc_url( '?source=' . $source ),
```

Reply:
Thanks for the suggestions. Makes sense making it more generic.